api/app/Http/Controllers/Api/AuthController.php

```diff
@@ -31,6 +36,28 @@ public function login(LoginRequest $request): JsonResponse
         return $this->respondWithItem($this->guard->user());
     }
 
+    public function register(RegisterRequest $request): JsonResponse
+    {
+        $role = Role::CONTRIBUTOR();
+        $name = $request->get('name');
+        $email = $request->get('email');
+        $password = bcrypt($request->get('password'));
+        $user = new User($role, $name, $email, $password);
+        if ($request->get('nickname')) {
+            $user->setNickname($request->get('nickname'));
+        }
+        $this->em->persist($user);
+        $this->em->flush();
+
+        if (!$this->guard->attempt($request->credentials(), false)) {
+            throw new AuthenticationException(__('auth.failed'));
+        }
```

Comment:
Instead of this, you should just do:
```suggestion
        $this->guard->login($user, false);
```